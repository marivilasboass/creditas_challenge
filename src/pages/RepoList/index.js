import React from 'react';
import {Text, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  welcome: {
    color: '#000',
    fontSize: 30,
    fontWeight: 'bold',
    textAlign: 'center',
    marginTop: '50%',
  },
});

const Main = () => (
  <Text style={styles.welcome}>LIST!</Text>
);

export default Main;
