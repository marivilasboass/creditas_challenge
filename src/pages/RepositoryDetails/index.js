import React, {useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';

import {getCommits, getCommitAvatar} from '~/store/ducks/repository';

import {
  Indicator,
  ScrollViewWrapper,
  Wrapper,
  Logo,
  Title,
  Post,
  Avatar,
  MessageWrapper,
  Message,
} from './style';

const RepositoryDetails: () => React$Node = ({navigation}) => {
  const dispatch = useDispatch();
  const repo = navigation.getParam('repo');
  const commitState = useSelector(state => state.repository.commits);
  const commits = commitState[repo.id];
  const isLoadingCommits =
    useSelector(state => state.repository.loadingCommits) || false;

  useEffect(() => {
    if (!commits) {
      if (dispatch) {
        dispatch(getCommits(repo));
      }
    }
  }, [commits, dispatch, repo]);

  if (isLoadingCommits || !commits) {
    return <Indicator />;
  }

  return (
    <Wrapper>
      <Logo source={require('~/assets/images/logo-full.png')} />
      <Title>
        {repo.name} ({commits.length} commits)
      </Title>
      <ScrollViewWrapper>
        {commits.map((commit, index) => (
          // eslint-disable-next-line react-native/no-inline-styles
          <Post key={commit.sha} style={{borderTopWidth: index ? 1 : 0}}>
            <Avatar source={getCommitAvatar(commit)} />
            <MessageWrapper>
              <Message>{commit.commit.message}</Message>
            </MessageWrapper>
          </Post>
        ))}
      </ScrollViewWrapper>
    </Wrapper>
  );
};

export default RepositoryDetails;
