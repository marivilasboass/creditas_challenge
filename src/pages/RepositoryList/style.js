import styled from 'styled-components/native';
import {Dimensions} from 'react-native';
import {theme} from '~/theme/globalStyle';

export const Wrapper = styled.View`
  flex: 1;
  margin-top: 20%;
  background: ${theme.color.white};
  align-items: center;
  padding-left: ${theme.defaultPadding.small};
  padding-right: ${theme.defaultPadding.small};
  justify-content: center;
`;

export const Logo = styled.Image`
  width: ${Dimensions.get('window').width};
  height: 80px;
  margin-bottom: ${theme.defaultPadding.large};
  resize-mode: contain;
  align-self: center;
`;

export const AnimatedWrapper = styled.Animated.View`
  height: 150px;
`;
