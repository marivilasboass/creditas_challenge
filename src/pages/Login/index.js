import React, {useState} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {Platform} from 'react-native';
import {Types, userLogin, userAddEmail} from '~/store/ducks/user';

import {Wrapper, Logo, FormWrapper, Input, LoginText} from './style';
import Button from '~/components/Button';

const Login: () => React$Node = ({navigation}) => {
  const userState = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [setFocus] = useState(false);
  const keyboardBehavior = Platform.OS === 'android' ? 'padding' : 'position';

  function onClickNext() {
    userAddEmail({dispatch})({email});
  }
  function onClickBack() {
    dispatch({type: Types.LOGIN_RESET});
    setFocus(false);
  }

  function onClickLogin() {
    userLogin({dispatch})({email: userState.email, password, navigation});
  }

  function onFocus() {
    setFocus(true);
  }

  function onBlur() {
    setFocus(false);
  }

  return (
    <Wrapper>
      <Logo source={require('~/assets/images/logo.png')} />
      <FormWrapper behavior={keyboardBehavior}>
        {userState.email ? (
          <>
            <LoginText>{userState.email}</LoginText>

            <Input
              placeholder="password"
              onChange={value => setPassword(value.nativeEvent.text)}
              secureTextEntry
              onFocus={onFocus}
              onBlur={onBlur}
              autoFocus
            />
            <Button title="login" onPress={onClickLogin} />
            <Button title="back" onPress={onClickBack} />
          </>
        ) : (
          <>
            <Input
              placeholder="login"
              onChange={value => setEmail(value.nativeEvent.text)}
              keyboardType="email-address"
              autoCapitalize="none"
              autoCompleteType="off"
              autoCorrect={false}
              onFocus={onFocus}
              onBlur={onBlur}
            />
            <Button title="next" onPress={onClickNext} />
          </>
        )}
      </FormWrapper>
    </Wrapper>
  );
};

export default Login;
