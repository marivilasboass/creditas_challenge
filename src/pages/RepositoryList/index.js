import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {StatusBar, Keyboard, Animated} from 'react-native';
import {connect} from 'react-redux';

import {configureLayoutAnimation} from '~/utils/layoutAnimation';
import {filterByKeyword} from '~/utils/repoFilter';
import {getRepos} from '~/store/ducks/repository';

import RepoListScrollView from './components/ListItemScroll';
import SearchInput from '~/components/SearchInput';
import {Wrapper, Logo} from './style';

export class RepositoryList extends PureComponent {
  static propTypes = {
    repos: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        description: PropTypes.string,
      }),
    ),
    getRepos: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
  };

  static defaultProps = {
    isLoading: false,
    repos: [],
    getRepos: () => {},
  };

  state = {
    keyword: '',
  };

  scrollYValue = new Animated.Value(0);

  componentDidMount() {
    this.props.getRepos();
  }

  onSearch(keyword) {
    clearTimeout(this.typeKeywordDebounce);

    this.typeKeywordDebounce = setTimeout(() => {
      configureLayoutAnimation();
      this.setState({keyword});
    }, 300);
  }

  onChange(value) {
    if (!value) {
      Keyboard.dismiss();
    }
    this.onSearch(value.nativeEvent ? value.nativeEvent.text : value);
  }

  render() {
    const {repos, isLoading} = this.props;
    const {keyword} = this.state;

    return (
      <Wrapper>
        <StatusBar barStyle="default" />
        <Animated.View>
          <Logo source={require('~/assets/images/logo-full.png')} />
          <SearchInput
            defaultValue={keyword}
            onChange={value => this.onChange(value)}
            placeholder="search"
          />
        </Animated.View>

        <RepoListScrollView
          isLoading={isLoading}
          getRepos={this.props.getRepos}
          repos={filterByKeyword(repos, keyword)}
          onScroll={Animated.event([
            {nativeEvent: {contentOffset: {y: this.scrollYValue}}},
          ])}
        />
      </Wrapper>
    );
  }
}

const mapStateToProps = state => ({
  repos: state.repository.repos || [],
  isLoading: state.repository.loading,
});

const mapDispatchToProps = {
  getRepos,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(RepositoryList);
