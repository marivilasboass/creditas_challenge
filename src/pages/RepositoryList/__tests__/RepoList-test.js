import 'react-native';
import React from 'react';
import {RepositoryList} from '~/pages/RepositoryList';
import renderer from 'react-test-renderer';

it('renders RepositoryList correctly (snapshot)', () => {
  const component = renderer.create(<RepositoryList />).toJSON();
  expect(component).toMatchSnapshot();
});
