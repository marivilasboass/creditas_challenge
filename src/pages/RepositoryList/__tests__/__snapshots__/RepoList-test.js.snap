// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RepoList correctly (snapshot) 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 20,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": 150,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/Assets/github-logo-full.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 60,
          "resizeMode": "contain",
        }
      }
    />
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#CCC",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 40,
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#CCC",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="off"
          autoCorrect={false}
          defaultValue=""
          onChange={[Function]}
          placeholder="Repository Search"
          placeholderTextColor="#333"
          rejectResponderTermination={true}
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CCC",
                  "fontSize": 24,
                },
                Object {
                  "opacity": 0,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    onScroll={[Function]}
    onScrollEndDrag={[Function]}
    scrollEventThrottle={16}
  >
    <View />
  </RCTScrollView>
</View>
`;
